Public/Src/Cache/ContentStore/Library/Utils/BandwidthChecker.cs

```diff
@@ -45,93 +45,78 @@ public async Task<CopyFileResult> CheckBandwidthAtIntervalAsync(OperationContext
         {
             if (_historicalBandwidthLimitSource != null)
             {
-                var startPosition = destinationStream.Position;
+                var startPosition = tryGetPosition(destinationStream, out var pos) ? pos : 0;
                 var timer = Stopwatch.StartNew();
+                var (result, lastPosition) = await impl();
+                timer.Stop();
 
-                try
-                {
-                    return await impl();
-                }
-                finally
-                {
-                    timer.Stop();
-                    var endPosition = destinationStream.Position;
+                // Bandwidth checker expects speed in MiB/s, so convert it.
+                var bytesCopied = lastPosition - startPosition;
+                var speed = bytesCopied / timer.Elapsed.TotalSeconds / (1024 * 1024);

```

Comment:
A minor thing, but what do you think about this approach:

1. We can change the result of impl to return bytesCopied and
2. Add an extension method to compute mbps:

in this case the code will looks like this:

var (result, bytesCopied) = await impl;
var spped = bytesCopied.ToMbps(timer.Elapsed);
_historicalBandwidthLimitSource.AddBandwidthRecord(speed);

Reply:
The extension method seems a little overkill to me. Otherwise, agree.